### DISCUSSION TITLE
Google Season Of Docs 2024 Proposal for JSON Schema

### DISCUSSION THREAD

**Discussion by benjagm** [json-schema-org/community#687] [General]
![gsod](https://github.com/json-schema-org/community/assets/40007659/3c21facd-62b1-49af-ba0e-db82a2d62af8)

# 2024 Google Season of Docs (GSoD) Application

Below is the project proposals we are planning for this year's GSoD 2024. Application deadline is April 2nd.

Join us in the `#documentation` channel in [Slack](https://json-schema.org/slack) to continue the discussion.
Stay tuned!

# About JSON Schema

JSON Schema is a vocabulary that allows you to annotate and validate JSON documents 

We are a community JSON Schema enthusiast dedicated to maintain, evolve and promote the JSON Schema specification. The Community consists of individuals, community members, tooling builders, schema designers, researchers, and representatives from companies and organizations who use or are considering using JSON Schema.

# Proposed JSON Schema Docs Projects

## Current JSON Schema Docs Problem

After 10 different releases over the last 15 years the JSON Schema specification has turned in a very complex document too much focused on tooling creators but difficult to understand for general JSON Schema users. To close this GAP, the community has been creating a set of different documentation resources including a book, specification release notes, and some examples however those resources have proven to be insufficient to provide a good onboarding and learning experience for developers.

We already started the work to improve this situation adopting the [diataxis](https://diataxis.fr/) documentation principles, defining our docs [vision](https://github.com/json-schema-org/website/issues/245) and [delivering the documentation strategy](https://github.com/json-schema-org/website/issues/158), however we need technical writing support to be able to solve the core problems of our docs.

### 🎯 Objective

The objective of this project is to revamp the JSON Schema [Specification docs](https://json-schema.org/specification) to improve discoverability, structure, consistency and usability of whole section without replacing the official JSON Schema releases.

![Screenshot 2024-04-01 at 13 10 12](https://github.com/json-schema-org/community/assets/40007659/a31dfa85-1ef0-40ed-a846-c4963b34a975)

This project has been identified in our docs strategy as [Release 4](https://github.com/json-schema-org/website/issues/158#release4).

### 🔎 The Problem

When launching the new website past year we focused on rebranding and the landing page however we didn't have the time to improve the core of our docs. We re-published the Specification docs within a new whole section but those docs lack structure and consistency.

### ✅ The Scope

This project will provide a new version of the Specification docs, with a better structure to make it easy to find the current version of the specification inviting to migrate, understand the evolution of JSON schema over the different releases and the features added in each release while highlighting the current version and providing a consistent structure for navigation and discoverability making this section useful for both schema creators and tooling implementers.

### 📈 Measuring success

To measure the impact of the project success will be measured through various metrics, including unique visitors, visits, page views and time on page. 

![Screenshot 2024-04-01 at 13 33 26](https://github.com/json-schema-org/community/assets/40007659/6d24d4bd-9719-4a56-bf79-1076b4da12eb)

Additionally, we are continuous monitoring of user feedback to measure the user's sentiment and identify improvements.

![Screenshot 2024-04-01 at 13 37 10](https://github.com/json-schema-org/community/assets/40007659/4e01ada1-ca83-4191-b99f-2517455b2e18)

### 🗓️ Timeline

The project will take approximately 4-6 months to complete and can and may involve 1 technical writer. 

| **Dates** | **Action Items**  |
| --- | ----------- |
| May | Onboarding and enablement |
| June | Audit existing documentation and identify improvement areas |
| July - September | Implement the new version of the Specification docs | 
| October | Wrap up and release notes |
| November | Project completion and GSoD contributors receive some swag |

### 🎯 Objective

The objective of this project is to rebuild the reference section of the JSON Schema docs to better enable new JSON Schema users to easily find and understand the core concepts of JSON Schema.

This project has been identified in our docs strategy as [Release 5](https://github.com/json-schema-org/website/issues/158#release5).

### 🔎 The Problem

The core resource of the JSON Schema user docs been the book "[Understanding JSON Schema](https://json-schema.org/UnderstandingJSONSchema.pdf)", donated to the project by Michael Droettboom and for years it server os well, however its book structure makes it difficult for users to find what they are looking for because they expect the docs structure as modern developer docs. In addition, this book structure makes it difficult for us to maintain and evolve our docs.

![Screenshot 2024-04-01 at 14 47 33](https://github.com/json-schema-org/community/assets/40007659/4370e2ee-fafe-435c-aaaf-4ae24048c9c2)
 
### ✅ The Scope

This project will provide a new version of the Reference docs organized as modern developer docs using the [diataxis](https://diataxis.fr/) documentation principles by replacing the existing "Understanding JSON Schema" book. The new section will provide a most efficient onboarding, a better organization of contents as well as better examples and styles.

### 📈 Measuring success

To measure the impact of the project success will be measured through various metrics, including unique visitors, visits, page views and time on page. 

![Screenshot 2024-04-01 at 14 54 25](https://github.com/json-schema-org/community/assets/40007659/23d007d2-f217-496f-8a36-08783e75cc87)

Additionally, we are continuous monitoring of user feedback to measure the user's sentiment and identify improvements.

![Screenshot 2024-04-01 at 13 37 10](https://github.com/json-schema-org/community/assets/40007659/4e01ada1-ca83-4191-b99f-2517455b2e18)

### 🗓️ Timeline

The project will take approximately 4-6 months to complete and can and may involve 1 technical writer. 

| **Dates** | **Action Items**  |
| --- | ----------- |
| May | Onboarding and enablement |
| June | Audit existing documentation and identify improvement areas |
| July - September | Implement the new version of the Reference docs | 
| October | Wrap up and release notes |
| November | Project completion and GSoD contributors receive some swag |

<a id="budget"></a>
## 💸 Total Budget

In case of being selected, JSON Schema would like to request a US $5,000 budget for each project. The total budget request would be $10,000 for both projects. That said, we would also like to request an additional $350 in our budget to produce cool swag to award the participants.

Budget item | Total Amount
-- | --
Technical writing for the Specification Project | $5,000
Technical writing for the Reference Project | $5,000
Swag to award selected participants | $350
TOTAL | $10,350
<a id="apply"/>

## How to apply

Please use [this form](https://forms.gle/rZo1rCeY3xYgNZ6u6) to submit your application. 

The application process goes as follows:

**The application process has been closed due to the high number of applications**

~~1. Submit your application until May 2nd.~~ 
~~2. Interviews : April 29th to May 10th.~~
~~3. Make sure you submit your [Project Statement](https://developers.google.com/season-of-docs/docs/tech-writer-statement#project_statement) before May 10th to [this](mailto:benjamin.granados@postman.com) email.~~
4. Announcement : May 14th.

We are so looking forward to collaborating with you!!

## Eligibility Criteria

While other organizations run GSoD as a mentoring program, that is not our case. We think it is important to support the Technical Writing Community and support that Technical Writing discipline, so our ideal target are professional Technical Writers.

We are looking for a resourceful Technical Writer with good planning and execution skills, with passion for developers, ideally experience working with other standards and at least 1 year of experience in Technical Writing.

We are not forcing a specific dedication of 10/20/30/40 weekly hours, but we expect this project being a priority for you. It is totally fine if you are working in different projects, but if you are working full time for an specific organization it is going to be hard getting the time required.

## Next steps

If you are considering applying to this position we encourage you to start [exploring our docs](https://json-schema.org/learn/getting-started-step-by-step) and getting some understanding about JSON Schema, and the current Developer Experience we provide. This way your onboarding will be quicker and you will get a sense of why we need someone like you to help us to elevate our docs.

**Comment by Abhijay007** [comment]
Hi @benjagm,

The Slack link is no longer active. Could you please share the active link so that I can join the discussion at `#documentation` channel? I've read through both aspects of the project and I have extensive experience with various documentation frameworks. I believe I possess the relevant experience needed to accomplish the project. I would like to discuss the possibilities further.

Thank you!

**Reply by benjagm** [reply]
Thanks for your interest @Abhijay007 . Can you share the issue you get with the slack link? It is supposed to work

https://json-schema.org/slack

**Reply by Apoorvgarg-creator** [reply]
It is not working @benjagm

**Reply by benjagm** [reply]
Can you please try again?

**Reply by Abhijay007** [reply]
Hi @benjagm it still not working

**Comment by Apoorvgarg-creator** [comment]
<img width="728" alt="Screenshot 2024-04-10 at 11 58 48 PM" src="https://github.com/json-schema-org/community/assets/57873504/75eb45f5-50bf-46ae-8aec-2e2c09c45dfe">

**Comment by yashpandey06** [comment]
@benjagm   kudos !!! to the team for getting selected in GSOD 2024 ...looking forward to healthy contributions.

**Comment by benjagm** [comment]
Please use this link: https://join.slack.com/t/json-schema/shared_invite/zt-2gew5jhhj-WNWmCC8QJ~oHJ4rfNEXddQ

**Reply by Abhijay007** [reply]
Thanks @benjagm

**Comment by himanshuaggar** [comment]
Hi @benjagm , I would love to contribute to the JSON Schema Documentation for Google Season of Docs 2024.

**Comment by Snehanaik369** [comment]
Hello @benjagm, I'm Sneha Naik, a technical writer with significant expertise in developing project documentation. As a final-year student, I've utilized JSON extensively in various projects and have a deep understanding of its workings. I'm eager to contribute to the JSON Schema Documentation for Google Season of Docs 2024. Thanks a lot for your time and consideration.

**Comment by benjagm** [comment]
Hi everyone! We just published the eligibility criteria and details about how to apply and the application process timelines.

I am so looking forward to speak with all of you!!

**Comment by Swpn0neel** [comment]
Hey,

I'm Swapnoneel Saha and am extremely interested in working on this project. I'm currently working as a Technical Writer at Tutorials Point, and have a lot of prior experience of writing technical content for my clients as part of my freelancing ventures. Also, this project aligns with my current expertise, and that's why I believe that I'll be a great fit for this project.

Here's my link to my [LinkedIn profile](https://www.linkedin.com/in/swapnoneel-saha-14a3161b6/)
And, here are some of my writing samples:
1) https://swapnoneel.hashnode.dev/javascript-objects-vs-json
2) https://swapnoneel.hashnode.dev/inheritance-in-python
3) https://swapnoneel.hashnode.dev/devbytes-short-coding-news

Thank You,
Swapnoneel Saha
swapnoneelsaha111@gmail.com

**Comment by Ayush9026** [comment]
Thanks @benjagm sir for publishing the eligibility criteria and details about how to apply and the application process timelines.

**Comment by benjagm** [comment]
Hi everyone! We just modified the application process to add a step to submit the Project Statement before May 10th:

The application process goes as follows:

1. Submit your application until May 2nd.
2. Interviews : April 29th to May 10th.
3. Make sure you submit your [Project Statement](https://developers.google.com/season-of-docs/docs/tech-writer-statement#project_statement) before May 10th to [this](mailto:benjamin.granados@postman.com) email.
4. Announcement : May 14th.

**Reply by nitishmalang** [reply]
@benjagm   email is not working for project statement

**Reply by benjagm** [reply]
> @benjagm email is not working for project statement

Try again please.

**Reply by nitishmalang** [reply]
yes now it's working

**Comment by kaushal07wick** [comment]
hey @benjagm is the submission period over. I was just going to submit the statement of interest. I saw you making a change right now.

**Reply by benjagm** [reply]
Hi @kaushal07wick . Thanks a lot for your interest. We closed the process today, that is correct.

We'd like to be mindful with everyone's time but at this point we think we have enough candidates to complete the process.

**Comment by benjagm** [comment]
Hi everyone! 

Due to the high number of applications we decided today to close the application process. We defined an application process giving everyone the chance to go to the screening interview with me, but it ended in a exhausting process that of course brought many benefits but also was highly demanding.

We'll take the decision with the applications we already received and the interviews we already had and we'll make announcement on the agreed date, May 14th.

Thanks a lot for your interest and flexibility.

**Reply by benjagm** [reply]
This has been a learning process for us and next time well plan for a more traditional approach, starting with a statement of interest and only interviewing the candidates that really have a good fit with the position, because what we tried interviewing everyone didn't work for a small team like us.

**Comment by Temidayo32** [comment]
> Hi everyone!
> 
> Due to the high number of applications we decided today to close the application process. We defined an application process giving everyone the chance to go to the screening interview with me, but it ended in a exhausting process that of course brought many benefits but also was highly demanding.
> 
> We'll take the decision with the applications we already received and the interviews we already had and we'll make announcement on the agreed date, May 14th.
> 
> Thanks a lot for your interest and flexibility.

Oh, wow.

**Comment by benjagm** [comment]
Hi everyone.

I wanted to send an additional message to apologize for not providing the best application experience in our first GSoD experience. We intended to provide a different and 100% human experience by interviewing with all the candidates with technical writing experience, however I didn't planed for such a high number of applications which ended up with 2 weeks full of interviews added to my current commitments. The process was exhausting but it gave us the chance to get to know to most of the applicants and in the process we found strong candidates to secure the success of the project. In that scenario I decided to be respectful with everyone's time still working in project statements and cancel few pending interviews and close the application process to focus on the last discussions.

I understand that this is not the best application experience, but the reality is that we are not a Company, we are an open source project with a small team with limited resources, and next time we'll make sure of deciding a more traditional process that we can manage without get burned. 

I'd like to take responsibility of this situation and any possible negative experience you may have interacting with us. That does not represent the JSON Schema Community and I hope we'll have opportunities to work together in this or any other Open Source project.

Sincerely.

Benjamin Granados
Community Manager

**Reply by nitishmalang** [reply]
@benjagm  we understand your concern you have to go through large number of applications. Its ok I also got rejected this time. But I will continue my work on contributing in open-source & come-back next time with strong application
Thank-you
